V1.1.0

vincerubinetti opened the pull request:
FOR THE TEMPLATE MAINTAINER(S)

Checklist:

- [x] I have updated CITATION and CHANGELOG as appropriate.
- [x] I have updated lab-website-template-docs as appropriate.
- [x] I have checked the testbed as appropriate.

falquaddoomi commented on .docker/run.sh:
```diff
@@ -0,0 +1,35 @@
+# choose platform
+PLATFORM=""
+
+# default docker run command
+DOCKER_RUN="docker run"
+
+# fix interactive on windows
+if [[ $OSTYPE == msys* ]]; then
+    DOCKER_RUN="winpty docker run"
+fi
+
+# build docker image
+docker build ${PLATFORM} \
+    --tag ${IMAGE} \
+    --file ./.docker/Dockerfile .
+
+# run built docker image
+${DOCKER_RUN} ${PLATFORM} \
+    --name ${CONTAINER} \
+    --init \
+    --rm \
+    --interactive \
+    --tty \
+    --publish 4000:4000 \
+    --publish 35729:35729 \
+    --volume $PWD:/usr/src/app \
+    ${IMAGE} "$@"
```
This version looks like it retains the issue where Docker for Windows gets a Linux-style path and doesn't know what to do with it. I've added a fix based on our conversation the other day, where you found that `cmd //c cd` worked without any modification.

FWIW, it seems that the value of `$OSTYPE` on Windows can vary depending on how you run the bash script. (I found a few values for the variable in this issue: https://github.com/dylanaraps/neofetch/issues/433.) For instance, running it through Git Bash produces `msys`, and through vanilla CYGWIN produces `cygwin`. Unfortunately running through the Windows 10's Windows Subsystem for Linux (WSL), it returns `linux-gnu`, which is the same as other platforms that do use Linux-style paths. I've modified the check for Windows to at least match `msys` and `cygwin`, but it doesn't handle the WSL `linux-gnu` value, which is conflated with real Linux anyway and definitely shouldn't use a Windows path in that case. I think the more robust solution here would be running `docker info` and extracting the platform the Docker daemon is running on, but we can wait on that. For now, I think these changes should be fine.

```suggestion
# default path to current working directory
APP_DIR="$PWD"

# default docker run command
DOCKER_RUN="docker run"

# fix interactive, folder mounting on windows
if [[ $OSTYPE == msys* ]] || [[ $OSTYPE == cygwin* ]]; then
    DOCKER_RUN="winpty docker run"
    APP_DIR="$( cmd //c cd )"
fi

# build docker image
docker build ${PLATFORM} \
    --tag ${IMAGE} \
    --file ./.docker/Dockerfile .

# run built docker image
${DOCKER_RUN} ${PLATFORM} \
    --name ${CONTAINER} \
    --init \
    --rm \
    --interactive \
    --tty \
    --publish 4000:4000 \
    --publish 35729:35729 \
    --volume "${APP_DIR}:/usr/src/app" \
    ${IMAGE} "$@"
```

falquaddoomi reviewed (commented):
Most of it looks great; I made a few comments on the Docker inclusions.

falquaddoomi reviewed (approved):
Nice, this is looking good now. I suppose you should try running it under real Windows before we merge, but so far, so good.

vincerubinetti commented:
Your previous review comments and changes that we incorporated for #162 were somehow either never committed or reverted, but luckily I restored them from my local clone.

Added a commit to tweak the OS detection, splitting the shell concern from the filesystem path concern. I don't think the `*` is needed in `$DOCKER_OSTYPE == windows`; it doesn't seem like `docker info -> OSType` includes any kind of version or variations.
